atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -1020,26 +1009,16 @@ private void fillOverflowValues(DbReadTable table,
                 overflowIterator.remove();
             }
         }
-
-        if (!overflowValues.isEmpty()) {
-            Map<Long, byte[]> resolvedOverflowValues = Maps.newHashMapWithExpectedSize(overflowValues.size());
-            try (ClosableIterator<AgnosticLightResultRow> overflowIter = table.getOverflow(overflowValues.values())) {
-                while (overflowIter.hasNext()) {
-                    AgnosticLightResultRow row = overflowIter.next();
-                    // QA-94468 LONG RAW typed columns ("val" in this case) must be retrieved first from the result set
-                    // see https://docs.oracle.com/cd/B19306_01/java.102/b14355/jstreams.htm#i1007581
-                    byte[] val = row.getBytes(VAL);
-                    long id = row.getLong("id");
-                    resolvedOverflowValues.put(id, val);
-                }
-            }
-            for (Entry<Cell, OverflowValue> entry : overflowValues.entrySet()) {
-                Cell cell = entry.getKey();
-                OverflowValue ov = entry.getValue();
-                byte[] val = resolvedOverflowValues.get(ov.id());
-                Preconditions.checkNotNull(val, "Failed to load overflow data: cell=%s, overflowId=%s", cell, ov.id());
-                values.put(cell, Value.create(val, ov.ts()));
-            }
+        Map<Long, byte[]> resolvedOverflowValues = overflowValueLoader.loadOverflowValues(
```

Comment:
We get rid of running the old and new queries lazily here. Not sure if that is a concern.